src/Microsoft.Framework.Logging.Console/ConsoleLoggerFactoryExtensions.cs

```diff
@@ -35,5 +37,23 @@ public static ILoggerFactory AddConsole(this ILoggerFactory factory, LogLevel mi
             factory.AddProvider(new ConsoleLoggerProvider((category, logLevel) => logLevel >= minLevel));
             return factory;
         }
+
+        /// <summary>
+        ///     Adds a console logger that is enabled as defined by the IConfiguration object
+        /// </summary>
+        /// <param name="configuration">IConfiguration object with filter attributes</param>
+        /// <returns></returns>
+        public static ILoggerFactory AddConsole(this ILoggerFactory factory, IConfiguration configuration)
+        {
+            factory.AddProvider(new ConsoleLoggerProvider((category, logLevel) =>
+            {
+                LogLevel minLevel;
+
+                return Enum.TryParse(configuration?.GetSubKey("ConsoleLogger")?.Get(category), out minLevel) &&
```

Comment:
@Eilon @lodejard  I won't know the `category` until the `CreateLogger` on the Provider is invoked. To read all categories before hand from configuration and cache it, I would need to get an `IEnumerable` from the Configuration object to parse each entry and maintain a Dictionary which I can pass to the Func. Do you think this is the right approach ? In that scenario, I need a method on IConfiguration something like `GetAllKey` so that I can iterate on them to read each value and perform the operation